elastic/kibana
# [Observability Onboarding] Prevent showing duplcated AWS services in Firehose flow

Closes #200931 

Switched to using AWS service list as a base to showing the detected services in the UI instead of the list of populated indices as multiple indices can be related to a single service.

### How to test

1. Go to Firehose flow `/observabilityOnboarding/firehose`
2. Open Kibana dev tools in another tab
3. Ingest documents related into multiple data streams which related to a single AWS service:
```
POST logs-aws.apigateway_logs-default/_doc
{
  "@timestamp": "2024-11-25T13:32:01.000Z",
  "some": 111,
  "aws.kinesis.name": "Elastic-CloudwatchLogs"
}

POST metrics-aws.apigateway_metrics-default/_doc
{
    "@timestamp": "2024-11-25T13:31:01.000Z",
    "agent": {
      "type": "firehose"
    },
    "aws": {
      "cloudwatch": {
        "namespace": "AWS/ApiGateway"
      },
      "exporter": {
        "arn": "arn:aws:cloudwatch:us-west-2:975050175126:metric-stream/Elastic-CloudwatchLogsAndMetricsToFirehose-CloudWatchMetricStream-Nhb4NhzPdL4J"
      }
    },
    "cloud": {
      "account": {
        "id": "975050175126"
      },
      "provider": "aws",
      "region": "us-west-2"
    }
}
```
4. Make sure you see only one entry for the service appear in the Firehose flow
